src/Cache.ts

```diff
@@ -39,12 +40,16 @@ export class Cache {
   public async putFileInCache(url: string, currentPath: string, fileName: string): Promise<string> {
     const cachePath = this.getCachePath(url, fileName);
     d(`Moving ${currentPath} to ${cachePath}`);
-    if (await fs.pathExists(cachePath)) {
-      d('* Replacing existing file');
-      await fs.remove(cachePath);
+
+    if (!fs.existsSync(path.dirname(cachePath))) {
```

Comment:
Note: this new clause is needed because `fs.rename` will fail if the target directory doesn't exist, while `fs.move` creates the directory.

This also doesn't work across disks, but we could catch the resulting error and fall back to an `fs.copy` call if necessary.